agent/internal/actor/http.go

```diff
@@ -120,13 +127,43 @@ func (p *blockedUserEventProperties) MarshalJSON() ([]byte, error) {
 func (p *blockedUserEventProperties) GetActionId() string {
 	return p.action.ActionID()
 }
-func (p *blockedUserEventProperties) GetOutput() api.BlockedUserEventProperties_Output {
-	return *api.NewBlockedUserEventProperties_OutputFromFace(p)
+func (p *blockedUserEventProperties) GetOutput() api.BlockedUserEventPropertiesOutput {
+	return *api.NewBlockedUserEventPropertiesOutputFromFace(p)
 }
 func (p *blockedUserEventProperties) GetUser() map[string]string {
 	return p.userID
 }
 
+// redirectedUserEventProperties implements `types.EventProperties` to be marshaled
+// to an SDK event property structure.
+type redirectedUserEventProperties struct {
+	action Action
+	userID map[string]string
+}
+
+// Static assert that redirectedUserEventProperties implements `types.EventProperties`.
+var _ types.EventProperties = &redirectedUserEventProperties{}
+
+func newRedirectedUserEventProperties(action Action, userID map[string]string) *redirectedUserEventProperties {
+	return &redirectedUserEventProperties{
+		action: action,
+		userID: userID,
+	}
+}
+func (p *redirectedUserEventProperties) MarshalJSON() ([]byte, error) {
+	pb := api.NewRedirectedUserEventPropertiesFromFace(p)
+	return json.Marshal(pb)
+}
+func (p *redirectedUserEventProperties) GetActionId() string {
+	return p.action.ActionID()
+}
+func (p *redirectedUserEventProperties) GetOutput() api.RedirectedUserEventPropertiesOutput {
+	return *api.NewRedirectedUserEventPropertiesOutputFromFace(p)
+}
+func (p *redirectedUserEventProperties) GetUser() map[string]string {
```

Comment:
This `map[string]string` shoud definitely be a dedicated type or struct. If there's only one it should ensure by code or structure that it is so.

Reply:
Same here - it implements an interface generated by go protobufs which corresponds to a field of a message in the API, so it matches the type of the API.